lnbits/app.py

```diff
@@ -105,52 +119,120 @@ def signal_handler(signal, frame):
+
+
+def check_installed_extension(ext: InstallableExtension) -> bool:
+    zip_files = glob.glob(
+        os.path.join(settings.lnbits_data_folder, "extensions", "*.zip")
+    )
+
+    if ext.has_installed_version:
+        return True
+    if ext.zip_path in zip_files:
+        ext.extract_archive()
+    else:
+        ext.download_archive()
+        ext.extract_archive()
+    return False
+
+
+async def restore_installed_extension(app: FastAPI, ext: InstallableExtension):
+    extension = Extension.from_installable_ext(ext)
+    register_ext_routes(app, extension)
+
+    current_version = (await db_versions()).get(ext.id, 0)
+    await migrate_extension_database(extension, current_version)
+
+    # mount routes for the new version
+    core_app_extra.register_new_ext_routes(extension)
+    if ext.module_installed:
+        ext.nofiy_upgrade()
+
+
 def register_routes(app: FastAPI) -> None:
     """Register FastAPI routes / LNbits extensions."""
     app.include_router(core_app)
     app.include_router(core_html_routes)
 
     for ext in get_valid_extensions():
         try:
-            ext_module = importlib.import_module(f"lnbits.extensions.{ext.code}")
-            ext_route = getattr(ext_module, f"{ext.code}_ext")
-
-            if hasattr(ext_module, f"{ext.code}_start"):
-                ext_start_func = getattr(ext_module, f"{ext.code}_start")
-                ext_start_func()
-
-            if hasattr(ext_module, f"{ext.code}_static_files"):
-                ext_statics = getattr(ext_module, f"{ext.code}_static_files")
-                for s in ext_statics:
-                    app.mount(s["path"], s["app"], s["name"])
-
-            logger.trace(f"adding route for extension {ext_module}")
-            app.include_router(ext_route)
+            register_ext_routes(app, ext)
         except Exception as e:
             logger.error(str(e))
             raise ImportError(
                 f"Please make sure that the extension `{ext.code}` follows conventions."
             )
 
 
+def register_new_ext_routes(app: FastAPI) -> Callable:
+    def register_new_ext_routes_fn(ext: Extension):
+        register_ext_routes(app, ext)
+
+    return register_new_ext_routes_fn
+
+
+def register_ext_routes(app: FastAPI, ext: Extension) -> None:
+    """Register FastAPI routes for extension."""
+    ext_module = importlib.import_module(ext.module_name)
+
+    ext_route = getattr(ext_module, f"{ext.code}_ext")
+
+    if hasattr(ext_module, f"{ext.code}_start"):
+        ext_start_func = getattr(ext_module, f"{ext.code}_start")
+        ext_start_func()
+
+    if hasattr(ext_module, f"{ext.code}_static_files"):
+        ext_statics = getattr(ext_module, f"{ext.code}_static_files")
+        for s in ext_statics:
+            app.mount(s["path"], s["app"], s["name"])
+
+    logger.trace(f"adding route for extension {ext_module}")
+
+    prefix = f"/upgrades/{ext.hash}" if ext.hash != "" else ""
```

Comment:
```suggestion
    prefix = f"/upgrades/{ext.hash}"
```

should do the same :)

Reply:
Not really, the new code would produce `"/upgrades/"` for prefix when the hash is empty.